Source/content/en/docs/tutorials/event-horizon.md

```diff
@@ -0,0 +1,501 @@
+            .fromProducerPartition(PartitionId.unspecified.value)
+            .toScope('808ddde4-c937-4f5c-9dc2-140580f6919e');
+        });
+})
+// Rest of builder here...
+
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+This is the most complicated piece of the code, and rightfully so there is a lot of things going on behind the scenes. Let's start at the 'event horizons' part.
+
+#### Event Horizon
+An event horizon is a subscription that is submitted to and managed in the Runtime. That subscription belongs to a tenant and it tells the Runtime that it wants to get events from a specific partition in a public stream of a tenant in another microservice (synonym with Runtime).
+
+On the consumer side when the Runtime receives a subscription like this it initiates a background process that will send a subscription request to the producer Runtime. If that request is acknowledged the producer Runtime will start a background process that sends over one event at a time.
+The consumer will receive events from the producer and put those events in a specialized event-log that is identified by the *scope* id. We'll talk more about the `scope` when we talk about the event handler.
+
+The acknowledgement part is important, and getting to it depends on two things:
+
+- The consumer needs to know where to access the other microservices, i.e the URL addresses.
+- The producer needs to give formal consent for a tenant in another microservice to subscribe to public streams of a tenant.
+
+We'll come back to how this is setup later.
+
+#### Scoped Event Handler
+So now that we have a consumer microservice that has subscribed to the producer's public stream `2c087657-b318-40b1-ae92-a400de44e507` the natural progression for us would be to register an event handler that handles those events. This code sequence also introduces a new way of creating event handlers. Previously we've made event handlers as classes in separated files, but the Client Builder gives us the opportunity to create these directly.
+This code will create a partitioned event handler with id `6c3d358f-3ecc-4c92-a91e-5fc34cacf27e`. That should feel familiar. The thing that may not be familiar is the *scope* part.
+Remember that the event horizon subscription includes a scope and that the events from the that event horizon gets put in a specialized event-log identified by the scope id. Having the scope id defined when creating an event handler signifies that it will only handle events that are put into that scope and no other. Read more about event handlers and scopes [here]({{< ref "docs/concepts/event_handlers_and_filters#scope" >}}).
+
+You can read more about the event horizon [here]({{< ref "docs/concepts/event_handlers_and_filters#scope" >}}).
+
+## Setup your environment
+Now that we have the different microservices setup with the code we need to setup the environment for them to run in and configure them to be connected to each other.
```

Comment:
```suggestion
Now we have the producer and consumer microservices [Heads]({{< ref "docs/concepts/overview#components" >}}) coded, we need to setup the environment for them to run in and configure their Runtimes to be connected.
```

I added the first mention of "Heads" here, could've done that earlier or maybe we can just talk about them as microservices?